Improving the section `next-builder` (#917)
zeit/docs: pages/docs/v2/deployments/official-builders/next-js-now-next.mdx

Using a custom server would require that all pages be routed through that custom server and the application would lose out on many of the benefits of [serverless](/docs/v2/deployments/concepts/lambdas/) Next.js. You can still achieve most of the logic that you have in a custom server by using [getInitialProps()](https://nextjs.org/docs/#fetching-data-and-component-lifecycle) and using [routes](/docs/v2/deployments/configuration/#routes).

> ### Exposing configuration

Reviewer: ```suggestion
### Exposing Configuration
```

> To support configuration, Next.js inlines the provided values into the Javascript bundle at build. This circumvents the performance concerns with runtime config along with the subtle issues associated with sharing config with the client.

Reviewer: ```suggestion
To support configuration, Next.js inlines the provided values into the JavaScript bundle at build. This circumvents the performance concerns with runtime configuration along with the subtle issues associated with sharing configuration with the client.
```

> You can add the `env` key in `next.config.js`:

Reviewer: ```suggestion
You can add the `env` key to the `next.config.js` file:
```

```js

> // next.config.js

Reviewer: ```suggestion
```

module.exports = {
  env: {
    customKey: process.env.customKey,
    mySecret: process.env.mySecret
  }
}
```

<Caption>

>   An example <InlineCode>next.config.js</InlineCode> configuration.

Reviewer: ```suggestion
  An example <InlineCode>next.config.js</InlineCode> file.
```

</Caption>

> Then, [expose the build env](/docs/v2/deployments/environment-variables-and-secrets) in your `now.json`:

Reviewer: ```suggestion
Then, [expose the build env](/docs/v2/deployments/environment-variables-and-secrets) in your `now.json` file:
```

```json
{
  ...
  "build": {
    "env": {
      "customKey": "value",
      "mySecret": "@my-secret-name"
    }
  }
}
```

<Caption>

>   An example <InlineCode>now.json</InlineCode> configuration file specifying a

Reviewer: ```suggestion
  An example <InlineCode>now.json</InlineCode> file specifying a
```

  build environment variable.
</Caption>

This will allow you to use `process.env.customKey` and `process.env.mySecret` in your code, for example:

```js

> // pages/index.js

Reviewer: ```suggestion
```

function Index() {
  return (
    <div>
      <h1>The value of customKey is: {process.env.customKey}</h1>
      <h2>Your secret is: {process.env.mySecret}</h2>
    </div>
  )
}

export default Index
```

> 

Reviewer: ```suggestion

<Caption>An example <InlineCode>index.js</InlineCode> file that references environment variables.</Caption>

```

Which will be replaced with:

```js

> // pages/index.js

Reviewer: ```suggestion
```

function Index() {
  return (
    <div>
      <h1>The value of customKey is: {'value'}</h1>
      <h2>Your secret is: {'secret-value'}</h2>
    </div>
  )
}

export default Index
```

> 

Reviewer: ```suggestion

<Caption>An example <InlineCode>index.js</InlineCode> file that shows the value of the referenced environment variables.</Caption>

```
